controllers/k8ssandra/k8ssandracluster_controller.go

```diff
@@ -602,21 +324,358 @@ func (r *K8ssandraClusterReconciler) setStatusForStargate(kc *api.K8ssandraClust
 	return nil
 }
 
-func (r *K8ssandraClusterReconciler) reconcileStargateAuthSchema(
-	ctx context.Context,
-	kc *api.K8ssandraCluster,
-	dc *cassdcapi.CassandraDatacenter,
-	remoteClient client.Client,
-	logger logr.Logger,
-) error {
-	managementApi, err := r.ManagementApi.NewManagementApiFacade(ctx, dc, remoteClient, logger)
-	if err == nil {
+func (r *K8ssandraClusterReconciler) checkDeletion(ctx context.Context, kc *api.K8ssandraCluster, logger logr.Logger) result.ReconcileResult {
```

Comment:
How about moving this method + `checkFinalizer` to a separate file, to keep this one short and concise? 

(We would need to do the same for Stargate-related methods and methods related to secrets.)